### PR TITLE
Promote Posts: Rename package to Blaze

### PR DESCRIPTION
**Pull request by dereksmart**
For the sake of consistency with the current branding, we're renaming the package to `blaze`. 

#### Changes proposed in this Pull Request:

No functional changes expected. 
New package: Blaze
Remove old package: Promote Posts

#### Other information:

This will need creation of a new mirror repo, and deploy to packagist. 

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

#### Jetpack product discussion

pedfti-Jk-p2

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

- Add this snippet to module-extras.php or a functionality plugin: 
```
use Automattic\Jetpack\Blaze;
Blaze::configure();
```

- Build the package: `jetpack build packages/blaze`
- Go to `/wp-admin/post-new.php`
- Look at the page source. `cmd+f` for `blaze/build/editor.js`. If it's there, it's loading!

### CODE REVIEW

File: projects/packages/blaze/src/class-blaze.php [hunk 1]
```diff
@@ -2,33 +2,33 @@
...
 	 */
 	public static function configure() {
-		$promote_posts = self::get_instance();
-		$promote_posts->register();
+		$blaze = self::get_instance();
+		$blaze->register();
 	}
 
 	/**
 	 * Initialize Promote Posts UIs.
```
Comment (coder-karen) [inline projects/packages/blaze/src/class-blaze.php]:
```suggestion
	 * Initialize Blaze UIs.
```

File: projects/packages/blaze/src/class-blaze.php [hunk 2]
```diff
@@ -61,22 +61,22 @@ public function register() {
 	 *
 	 * @return array
 	 */
-	public function jetpack_promote_posts_row_action( $post_actions, $post ) {
+	public function jetpack_blaze_row_action( $post_actions, $post ) {
 		$post_id = $post->ID;
 
 		// @todo wrap in method call, for general use?
 		$promote_post_url = Redirect::get_url(
```
Comment (coder-karen) [inline projects/packages/blaze/src/class-blaze.php]:
```suggestion
		$blaze_url = Redirect::get_url(
```

File: projects/packages/blaze/src/class-blaze.php [hunk 3]
```diff
@@ -61,22 +61,22 @@ public function register() {
...
-			'jetpack-promote-posts',
+			'jetpack-blaze',
 			array(
 				'query' => 'blazepress-widget=post-' . esc_attr( $post_id ),
 			)
 		);
 
 		// Add the link, make sure to tooltip hover.
-		$text  = _x( 'Promote', 'Verb', 'jetpack-promote-posts' );
+		$text  = _x( 'Promote', 'Verb', 'jetpack-blaze' );
```
Comment (coder-karen) [inline projects/packages/blaze/src/class-blaze.php:76]: Should this and the following printed wording be 'promote', or 'blaze'?

File: projects/packages/blaze/src/class-blaze.php [hunk 4]
```diff
@@ -61,22 +61,22 @@ public function register() {
...
 		$title = _draft_or_post_title( $post );
 		/* translators: post title */
-		$label                   = sprintf( __( 'Promote &#8220;%s&#8221; via Jetpack Social', 'jetpack-promote-posts' ), $title );
+		$label                   = sprintf( __( 'Promote &#8220;%s&#8221; via Jetpack Social', 'jetpack-blaze' ), $title );
 		$post_actions['promote'] = sprintf(
```
Comment (coder-karen) [inline projects/packages/blaze/src/class-blaze.php]:
```suggestion
		$post_actions['blaze'] = sprintf(
```

Also here, not sure how strict we want to be on the name.

File: projects/packages/blaze/src/class-blaze.php [hunk 5]
```diff
@@ -61,22 +61,22 @@ public function register() {
...
 			'<a href="%1$s" target="_blank" title="%2$s" aria-label="%2$s">%3$s</a>',
 			esc_url( $promote_post_url ),
```
Comment (coder-karen) [inline projects/packages/blaze/src/class-blaze.php]:
```suggestion
			esc_url( $blaze_url ),
```
...only if the variable name is changed on line 68.

File: projects/packages/blaze/src/js/editor.js [hunk 6]
```diff
@@ -44,7 +44,7 @@ const BlazePostPublishPanel = () => {
 		postVisibility: selector( editorStore ).getEditedPostVisibility(),
 	} ) );
 
-	const promoteUrl = getRedirectUrl( 'jetpack-promote-posts', {
+	const promoteUrl = getRedirectUrl( 'jetpack-blaze', {
```
Comment (coder-karen) [inline projects/packages/blaze/src/js/editor.js]:
```suggestion
	const blazeUrl = getRedirectUrl( 'jetpack-blaze', {
```

**Review by coder-karen — COMMENTED** [review]
Found a few suggested name changes throughout, though I'm not sure how strict the naming change should be in some cases.

File: projects/packages/blaze/src/class-blaze.php (hunk 3 above)
Comment (dereksmart) [reply projects/packages/blaze/src/class-blaze.php:76]: Will update the UI in a following PR, thanks!

File: projects/packages/blaze/src/class-blaze.php (hunk 2 above)
Comment (dereksmart) [reply projects/packages/blaze/src/class-blaze.php]: Reminds me the redirect-url needs updating in wpcom.

**Review by coder-karen — APPROVED** [review]

**Pull request merged**